var/spack/repos/builtin/packages/caliper/package.py

```diff
@@ -46,13 +50,18 @@ class Caliper(CMakePackage):
     variant('sosflow', default=False,
             description='Enable SOSflow support')
 
-    depends_on('gotcha@1.0:', when='+gotcha')
-    depends_on('dyninst', when='+dyninst')
-    depends_on('papi', when='+papi')
-    depends_on('libpfm4', when='+libpfm')
+    depends_on('gotcha@1.0.2:1.0.99', when='@1.0:1.99+gotcha')
+
+    depends_on('dyninst@9.3.0:9.3.2', when='@1.0:1.99+dyninst')
+
+    depends_on('papi@5.3.0:5.6.0', when='@1.0:1.99+papi')
```

Comment:
I think we can safely use anything with major version number 5 here. Same goes for libpfm (well, 4 there). Changes to the major version number should be the only thing to be concerned about. Plus perhaps a minimum version if older ones are broken, like dyninst.